### PR TITLE
Week 6 homework

### PR DESCRIPTION
**Pull request by CodiD**
This is my homework for week 6. I did work an the comments you made for week 5. I finished half of them and cleaned up mu code. I will be continuing to work on the rest.

### CODE REVIEW

File: Week6/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    def room_version = "2.2.5"
+
+    implementation "androidx.room:room-runtime:$room_version"
+    annotationProcessor "androidx.room:room-compiler:$room_version" // For Kotlin use kapt instead of annotationProcessor
+
+    // optional - Kotlin Extensions and Coroutines support for Room
+    implementation "androidx.room:room-ktx:$room_version"
+
+    // optional - RxJava support for Room
+    implementation "androidx.room:room-rxjava2:$room_version"
```
Comment (filbabic) [inline Week6/app/build.gradle:49]: You don't need these (Guava, Rx)

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/db/DaoMovie.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    @Insert(onConflict = OnConflictStrategy.IGNORE)
+    suspend fun insert(newMovie: MovieDataList)
+
+    @Query("DELETE FROM datalist")
+    suspend fun deleteAll()
+
+}
+
+
+
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/db/DaoMovie.kt:24]: .

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/db/MovieDatalist.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+class MovieDataList(
+    @PrimaryKey(autoGenerate = true) val id: Int,
+    @ColumnInfo(name = "releaseDate") val releaseDate: String,
+    @ColumnInfo(name = "title") val title: String,
+    @ColumnInfo(name = "summary") val summary: String,
+    @ColumnInfo(name = "genre") val genre: String,
+    @ColumnInfo(name = "poster") val poster: Int
+)
+
+
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/db/MovieDatalist.kt:18]: .

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/login/LoginActivity.kt [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+        val prefs = getSharedPreferences(PREFERENCE_NAME, Context.MODE_PRIVATE)
+        prefs.apply {
+            val userName = getString(NAME, "")
+            val password = getString(PASSWORD, "")
+            etUserName.setText(userName)
+            etPassword.setText(password)
+        }
+
+        btnLogin.setOnClickListener {
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/login/LoginActivity.kt:30]: naming

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/login/LoginActivity.kt [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    fun saveLoginData() {
+        val prefs = getSharedPreferences(PREFERENCE_NAME, Context.MODE_PRIVATE)
+        val editor = prefs.edit()
+            .putString(NAME, etUserName.text.toString())
+            .putString(PASSWORD, etPassword.text.toString())
+            .apply()
+    }
+
+    fun showLoginMessage() {
+        var loggedin = false
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/login/LoginActivity.kt:51]: .

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/login/LoginActivity.kt [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        loggedin = true
+    }
+
+    fun startMainScreen() {
+        val intent = Intent(applicationContext, MainActivity::class.java)
+        startActivity(intent)
+    }
+}
+
+private const val PREFERENCE_NAME = ""
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/login/LoginActivity.kt:64]: companion object or another file for constants

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/moviedetails/DetailActivity.kt [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_detail)
+        assigningValues()
+    }
+
+    fun assigningValues() {
+        movies = intent.getParcelableExtra(MainActivity.INTENT_LIST_KEY) as Movie
+        title = movies.title
+        tvTitle.text = movies.title
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/moviedetails/DetailActivity.kt:30]: .

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/MainActivity.kt [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+/** Main activity of the list page
+ *
+ */
+
+class MainActivity : AppCompatActivity(),
+    MovieAdapter.MovieListClickListener {
+
+    private lateinit var movieListRecyclerView: RecyclerView
+    private var movieDataManager =
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/MainActivity.kt:21]: val

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/MainActivity.kt [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+import com.isabelledionisius.mymovieapp.movies.model.Movie
+
+/** Main activity of the list page
+ *
+ */
+
+class MainActivity : AppCompatActivity(),
+    MovieAdapter.MovieListClickListener {
+
+    private lateinit var movieListRecyclerView: RecyclerView
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/MainActivity.kt:20]: why lateinit?

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/list/MovieListViewHolder.kt [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.isabelledionisius.mymovieapp.R
+
+// passing itemview to the superclass constructor
+class MovieListViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
+    /**
+     * View holder items for MovieAdapter class
+     */
+    var titleTextView = itemView.findViewById<TextView>(R.id.title)
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/list/MovieListViewHolder.kt:14]: extensions?

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/model/Movie.kt [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    val title: String,
+    val summary: String,
+    val genre: String,
+    val poster: Int
+) : Parcelable {
+    // I am not dealing with file descriptors => providing 0 value
+    override fun describeContents() = 0
+ }
+
+
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/model/Movie.kt:20]: .

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/model/MovieViewModel.kt [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+class MovieViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val repository: MovieRepository
+    // - We can put an observer on the data (instead of polling for changes) and only update the
+    //   the UI when the data actually changes.
+    // - Repository is completely separated from the UI through the ViewModel.
+    val movieItems: LiveData<List<MovieDataList>>
+
+    init {
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/model/MovieViewModel.kt:22]: again, through constructor

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/model/MovieViewModel.kt [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    suspend fun insert(movieItems: MovieDataList) {
+        // IO dispatcher as a context argument
+        // the dipatcher is used to define threads for coroutines
+        // context in general is a set of rules
+        coroutineScope {
+            launch (context = Dispatchers.IO) {
+            repository.movieRepository.insert(movieItems)
+            // posting back to the main thread
+            launch(Dispatchers.Main) {
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/movies/model/MovieViewModel.kt:39]: don't need this if you're not posting anything to the UI thread

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/repository/MovieRepository.kt [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+class MovieRepository(private val movieDao: DaoMovie) {
+
+    var movieRepository =
+        MovieRepository(
+            movieDao
+        )
+
+    // Declares the DAO as a private property in the constructor. Pass in the DAO
+// instead of the whole database, because you only need access to the DAO
+    class MovieRepository(private val movieDao: DaoMovie) {
```
Comment (filbabic) [inline Week6/app/src/main/java/com/isabelledionisius/mymovieapp/repository/MovieRepository.kt:16]: I commented on this before, but I guess it didn't go through.

No need to wrap a wrapper of the DAO into another layer. Also why have two of the same class names, with the same properties?

**Review by filbabic — CHANGES_REQUESTED** [review]
Your grade is "Good Job" here.

I keep saying this, but I'll have to be more strict about feedback being neglected. Most of the comments I've had, I've posted recently on the other PR; and none of them have been addressed.

Please fix these in the future.

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/repository/MovieRepository.kt (hunk 14 above)
Comment (CodiD) [reply Week6/app/src/main/java/com/isabelledionisius/mymovieapp/repository/MovieRepository.kt:16]: I have deleted the duplicated class. However, I am not quite sure how to change that (class MovieRepository(private val movieDao:DaoMovie)

Comment (filbabic) [reply Week6/app/src/main/java/com/isabelledionisius/mymovieapp/repository/MovieRepository.kt:16]: Not sure what you mean. It's fine to have `class MovieRepository(private val movieDao...). But you don't need another class with the same name, doing the same thing.

Comment (CodiD) [reply Week6/app/src/main/java/com/isabelledionisius/mymovieapp/repository/MovieRepository.kt:16]: OK. Then I got you, just deleted that

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/moviedetails/DetailActivity.kt (hunk 7 above)
Comment (CodiD) [reply Week6/app/src/main/java/com/isabelledionisius/mymovieapp/moviedetails/DetailActivity.kt:30]: What do you want me to do here? I just see a ., but no comments. Thank you

Comment (filbabic) [reply Week6/app/src/main/java/com/isabelledionisius/mymovieapp/moviedetails/DetailActivity.kt:30]: The . is used to just notify you of a comment I previously mentioned.

Here it's the `tvTitle` regarding naming, it should be `textViewTitle` or `titleView` or `movieTitle`.

File: Week6/app/src/main/java/com/isabelledionisius/mymovieapp/db/MovieDatalist.kt (hunk 3 above)
Comment (CodiD) [reply Week6/app/src/main/java/com/isabelledionisius/mymovieapp/db/MovieDatalist.kt:18]: What do the dots in the comments mean?

**Pull request merged**